mne/time_frequency/tfr.py

```diff
@@ -1138,15 +1179,310 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
+        ch_types = _get_channel_types(tfr.info)
+
+        # if multiple sensor types: one plot per channel type, recursive call
+        if len(ch_types) > 1:
+            logger.info("Multiple channel types selected, returning one "
+                        "figure per type.")
+            figs = list()
+            for this_type in ch_types:  # pick corresponding channel type
+                type_picks = [idx for idx in range(tfr.info['nchan'])
+                              if channel_type(tfr.info, idx) == this_type]
+                tf_ = _pick_inst(tfr, type_picks, None, copy=True)
+                if len(_get_channel_types(tf_.info)) > 1:
+                    raise RuntimeError(
+                        'Possibly infinite loop due to channel selection '
+                        'problem. This should never happen! Please check '
+                        'your channel types.')
+                figs.append(
+                    tf_.plot_joint(
+                        timefreqs=timefreqs, picks=None, baseline=baseline,
+                        mode=mode, tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax,
+                        vmin=vmin, vmax=vmax, cmap=cmap, dB=dB,
+                        colorbar=colorbar, show=False, title=title,
+                        layout=layout, yscale=yscale, combine=combine,
+                        exclude=None, topomap_args=topomap_args,
+                        verbose=verbose))
+            return figs
+
+        # Handle timefreqs
+        timefreqs = _get_timefreqs(tfr, timefreqs)
+        n_timefreqs = len(timefreqs)
+
+        ##############
+        # Image plot #
+        ##############
+
+        fig, tf_ax, map_ax, cbar_ax = _prepare_joint_axes(n_timefreqs)
+
+        cmap = _setup_cmap(cmap)
+
+        # image plot
+        # we also use this to baseline and truncate (times and freqs)
+        # (a copy of) the instance
+        fig = tfr._plot(
+            picks=None, baseline=baseline, mode=mode, tmin=tmin, tmax=tmax,
+            fmin=fmin, fmax=fmax, vmin=vmin, vmax=vmax, cmap=cmap, dB=dB,
+            colorbar=False, show=False, title=title, axes=tf_ax, layout=layout,
+            yscale=yscale, combine=combine, exclude=None, copy=False)
+
+        # set and check time and freq limits ...
+        # can only do this after the tfr plot because it may change these
+        # parameters
+        tmax, tmin = tfr.times.max(), tfr.times.min()
+        fmax, fmin = tfr.freqs.max(), tfr.freqs.min()
+        for time, freq in timefreqs.keys():
+            if not (tmin <= time <= tmax):
+                error_value = "time point (" + str(time) + " s)"
+            elif not (fmin <= freq <= fmax):
+                error_value = "frequency (" + str(freq) + " Hz)"
+            else:
+                continue
+            raise ValueError("Requesting a " + error_value + " not present "
+                             "in the data. Choose different `timefreqs`.")
```

Comment:
"I would prefer "Requested time point (...) is not present in the data" than "Requesting a time point (...) not present in the data."